web-console/src/views/query-view/query-view.tsx

```diff
@@ -251,17 +244,16 @@ export class QueryView extends React.PureComponent<QueryViewProps, QueryViewStat
         if (QueryView.isJsonLike(queryString)) {
           jsonQuery = Hjson.parse(queryString);
         } else {
-          const actualQuery = QueryView.wrapInLimitIfNeeded(queryString, wrapQueryLimit);
-
           jsonQuery = {
-            query: actualQuery,
+            query: queryString,
             resultFormat: 'array',
             header: true,
           };
         }
 
-        if (!isEmptyContext(queryContext)) {
+        if (!isEmptyContext(queryContext) || wrapQueryLimit) {
           jsonQuery.context = Object.assign(jsonQuery.context || {}, queryContext);
```

Comment:
May be better to do:

```js
jsonQuery.context = Object.assign({}, jsonQuery.context || {}, queryContext);
```

So there is no chance of changing something silly in pleace

Reply:
changed